bolt/include/bolt/Passes/IdenticalCodeFolding.h

```diff
@@ -27,15 +28,74 @@ class IdenticalCodeFolding : public BinaryFunctionPass {
       return false;
     if (BF.hasSDTMarker())
       return false;
+    if (BF.hasAddressTaken())
```

Comment:
We should place the "safe ICF" check here assuming `AddressTaken` property is independent of the ICF pass.